lib/routers/attachment.js

```diff
@@ -0,0 +1,77 @@
+const { Router } = require('express');
+const Busboy = require('busboy');
+const crypto = require('crypto');
+const { v4: uuid } = require('uuid');
+const sharp = require('sharp');
+
+const { S3 } = require('@asl/service/clients');
+const { NotFoundError } = require('@asl/service/errors');
+
+module.exports = settings => {
+
+  const router = new Router();
+  const s3 = S3(settings);
+  const { Attachment } = settings.models;
+
+  router.post('/', async (req, res, next) => {
+    const busboy = Busboy({ headers: req.headers, filesLimit: 1 });
+    const id = uuid();
+    const token = crypto.randomBytes(64).toString('hex');
+    const transform = sharp().resize(1200, undefined, { withoutEnlargement: true });
+    try {
+      const file = await new Promise((resolve, reject) => {
+        busboy.on('file', (field, Body, file) => {
+          if (file.mimeType.match(/^image\//)) {
```

Comment:
As a general attachments service would it be better to push the image resize logic out to the caller?